Applied-Upstream: commit: 6c759b422445a047d714f5b92cac8c6237e41033
Author: Campbell Jones <git@serebit.com>
Last-Update: 2022-03-07
Description: Add a null check to budgie_popover_manager_ungrab
 In rare cases, such as one I just experienced, this can cause the panel to segfault if the
 popover parameter is null. Better safe than sorry.
---
 src/plugin/popover-manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugin/popover-manager.c b/src/plugin/popover-manager.c
index a83f0423..080b0480 100644
--- a/src/plugin/popover-manager.c
+++ b/src/plugin/popover-manager.c
@@ -347,7 +347,7 @@ static void budgie_popover_manager_grab(BudgiePopoverManager* self, BudgiePopove
 static void budgie_popover_manager_ungrab(BudgiePopoverManager* self, BudgiePopover* popover) {
 	GdkDisplay* display = NULL;
 
-	if (!self->priv->grabbed || popover != self->priv->active_popover) {
+	if (popover == NULL || !self->priv->grabbed || popover != self->priv->active_popover) {
 		return;
 	}
 
-- 
2.34.1

